operator/api/vectorized/v1alpha1/cluster_types.go

```diff
@@ -222,6 +222,10 @@ type NodePoolSpec struct {
 	// Cache directory that will be mounted for Cloud Storage.
 	// +required
 	CloudCacheStorage StorageSpec `json:"cloudCacheStorage"`
+	// Additional command line arguments that we pass to the redpanda binary
+	// These are applied last and will override any other command line arguments that may be defined,
+	// including the ones added when setting `DeveloperMode` to `true`.
```

Comment:
I would suggest to deprecate https://github.com/redpanda-data/redpanda-operator/blob/4976e0732a16cc31e370ecee0c78d07b9f507e68/operator/api/vectorized/v1alpha1/cluster_types.go#L614-L617

Reply:
NodePools support is completely optional - the default NodePool controlled via top-level spec fields is still perfectly usable. therefore, it's not deprecated.

This shim here inits the default nodepool internal struct: https://github.com/redpanda-data/redpanda-operator/blob/jb%2Fnodepools/operator/api/vectorized/v1alpha1/cluster_types.go#L1436